llvm/test/Transforms/LoopVectorize/vector-to-scalar-cast.ll

```diff
@@ -0,0 +1,32 @@
+; RUN: opt --force-widen-divrem-via-safe-divisor=false -passes=loop-vectorize -S --debug-only=loop-vectorize < %s 2>&1 | FileCheck %s
```

Comment:
Missing REQUIRES: asserts, since we're checking debug strings.

Reply:
Would be good to wrap this line like:

```
RUN: opt ... \
RUN:   ... | Filecheck %s
```

Reply:
Is the `--force-widen-divrem-via-safe-divisor=false` needed?

Reply:
> I was able to reproduce it without this flag on some targets, but it seemed more robust to just force the code path that triggers the assert rather than depend on the target specific heuristics which I assume could be changed at any time.

Sounds good. +1 for making it target-independent.

Reply:
Actually, instead of checking debug strings, you can simply check the output (and it will be evident that the example is not vectorized), which can be generated using update_test_checks.py, and add a comment saying that the test was triggering an assertion earlier. That way, we don't have to rely on assertion builds (hence, no need for REQUIRES: asserts).

Reply:
Thanks, I have added such checks now, let me know what you think.

Reply:
Kindly remove `--debug-only=loop-vectorize`, as this will fail in release builds. Also, `2>&1` can be stripped.

Reply:
Ah of course, sorry I missed that, fixed now.